controllers/movement.py

```diff
@@ -38,6 +40,7 @@ def __init__(self) -> None:
         self.drive_local = False
 
         self.goal = Pose2d(math.inf, math.inf, math.inf)
+        self.waypoints: Optional[list[Translation2d]] = None
```

Comment:
Instead of making this optional why not just make it a list?

Reply:
i believe the type checker complained about an empty list being a mutable default, maybe a tuple will work?

Reply:
i believe this was just made to match the type of the waypoints parameter in set_goal which was an Optional because the type checker complained about a empty list being a mutable default, maybe it should be a tuple?

Reply:
Maybe this should just be any Sequence?